docs/deployment-patterns/deployment-as-code.md

```diff
@@ -0,0 +1,113 @@
+---
+title: Deployment as code
+description: You can define your entire Octopus configuration as code and track it all through source control. This page describes the different options available right now, and our plans to make this even better in the future.
+position: 4
+---
+
+You are probably reading this page because you want to manage your **deployment as code**. You want to define your deployment process, scripts, and variables in code, perhaps colocating this with the application source code in your git repository. This page describes the different options available in Octopus right now, and our plans to make this even better in the future.
+
+We recommend taking a two-phase approach when moving towards **deployment as code**:
+
+1. Start with [scripts as code](#scripts-as-code) - this method offers the best cost/reward ratio, is the simplest to implement and maintain over time.
+2. Move towards [project as code](#project-as-code) - this method is more difficult to set up, but comes with the benefit of having your deployment managed as code.
+
+## Scripts as code {#scripts-as-code}
+
+The simplest way to get started with **deployment as code** is to manage your custom deployment **scripts as code**. When you deploy your application, Octopus can execute a script contained inside a package. You can colocate your deployment scripts with your application source code, leveraging all the benefits of source control including change tracking and branching, then package it all up for Octopus.
+
+There is a downside to stopping here: your scripts are managed as code, but your deployment process and variables are still controlled by Octopus. Depending on your situation, this tradeoff might be quite acceptable.
+
+### Getting started
+
+You can follow this process to move your custom scripts without interrupting deployments:
+
+1. Move your deployment scripts from your Octopus deployment process into a file in your application source code
+1. Build your application as normal, but this time with the deployment scripts packaged up alongside your application
+  a. Your deployments should continue to work using the scripts in Octopus
```

Comment:
This feels odd. It "should", but it might not? 
Might be best to remove this line

Reply:
We should avoid should ;) it conveys uncertainty to users who are already probably feeling uncertain and have turned to documentation for reassurance/instruction.